go-controller/pkg/node/node.go

```diff
@@ -332,3 +344,187 @@ func buildEndpointAddressMap(epSubsets []kapi.EndpointSubset) map[epAddressItem]
 
 	return epMap
 }
+
+//watchPods watch updates for pod smart nic annotations
+func (n *OvnNode) watchPods() {
+	var retryPods sync.Map
+	// servedPods tracks the pods that got a VF
+	var servedPods sync.Map
+	_ = n.watchFactory.AddPodHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			klog.Infof("AddFunc:")
+			pod := obj.(*kapi.Pod)
+			if !ovn.PodWantsNetwork(pod) || pod.Status.Phase == kapi.PodRunning {
+				return
+			}
+			if ovn.PodScheduled(pod) {
+				// Is this pod created on same node where the smart NIC
+				if n.name != pod.Spec.NodeName {
```

Comment:
no, even when ovn-kubernetes run on the Bluefield the nodeName here will be the host because the context of this cni is to enable the Tenant Kubernetes, The other k8s cluster which will manage the BF will not use ovn-kubernetes cni but other cni (which we didn't define) as it doesn't expose VF and you don't need to manage the host.